src/strategies/polygon/strategy-sushi-farm-base.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7;
+
+import "./strategy-base.sol";
+import "../../interfaces/minichefv2.sol";
+
+abstract contract StrategySushiFarmBase is StrategyBase {
+    // Token addresses
+    address public constant sushi = 0x0b3F868E0BE5597D5DB7fEB59E1CADBb0fdDa50a;
+    address public constant miniChef = 0x0769fd68dFb93167989C6f7254cd0D766Fb2841F;
+
+    // WETH/<token1> pair
+    address public token0;
+    address public token1;
+
+    // How much SUSHI tokens to keep?
+    uint256 public keepSUSHI = 0;
+    uint256 public constant keepSUSHIMax = 10000;
+
+    uint256 public poolId;
+
+    constructor(
+        address _token0,
+        address _token1,
+        uint256 _poolId,
+        address _lp,
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyBase(
+            _lp,
+            _governance,
+            _strategist,
+            _controller,
+            _timelock
+        )
+    {
+        poolId = _poolId;
+        token0 = _token0;
+        token1 = _token1;
+        IERC20(sushi).safeApprove(sushiRouter, uint(-1));
+        IERC20(weth).safeApprove(sushiRouter, uint(-1));
+        IERC20(wmatic).safeApprove(sushiRouter, uint(-1));
```

Comment:
don't use this infinite approval.

Reply:
removed it, I think it should be also fixed in the original mainnet farm strategies too